activerecord/lib/active_record/railtie.rb

```diff
@@ -26,6 +26,9 @@ class Railtie < Rails::Railtie # :nodoc:
     config.active_record.use_schema_cache_dump = true
     config.active_record.maintain_test_schema = true
 
+    config.active_record_sqlite3adapter = ActiveSupport::OrderedOptions.new
+    config.active_record_sqlite3adapter.represent_boolean_as_integer = nil
```

Comment:
Okay, `active_record_sqlite3adapter` sounds like our internals bursting forth into people's apps, but if we just stash it in `config.active_record` we'll need the other code back.

How about:

`config.active_record.sqlite3.represent_boolean_as_integer = nil`